task.ps1

```
@@ -4,14 +4,17 @@ $resourceGroupName = "mate-azure-task-15"
 $virtualNetworkName = "todoapp"
 $vnetAddressPrefix = "10.20.30.0/24"
 $webSubnetName = "webservers"
-$webSubnetIpRange = "???" # <- calculate subnet ip address range 
+$webSubnetIpRange = "10.20.30.0/26"
 $dbSubnetName = "database"
-$dbSubnetIpRange = "???" # <- calculate subnet ip address range 
+$dbSubnetIpRange = "10.20.30.64/26"
 $mngSubnetName = "management"
-$mngSubnetIpRange = "???" # <- calculate subnet ip address range 
+$mngSubnetIpRange = "10.20.30.128/26"
 
 Write-Host "Creating a resource group $resourceGroupName ..."
 New-AzResourceGroup -Name $resourceGroupName -Location $location
 
 Write-Host "Creating a virtual network ..."
-# write your code here -> 
+$webservers = New-AzVirtualNetworkSubnetConfig -Name $webSubnetName -AddressPrefix $webSubnetIpRange
+$database = New-AzVirtualNetworkSubnetConfig -Name $dbSubnetName -AddressPrefix $dbSubnetIpRange
+$management = New-AzVirtualNetworkSubnetConfig -Name $mngSubnetName -AddressPrefix $mngSubnetIpRange
+New-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix $vnetAddressPrefix -Subnet $webservers ,$database, $management
```

The script assumes the Az module is installed and the user is authenticated (Connect-AzAccount). Those prerequisites are mentioned in the task description but not validated here. If you want clearer failures and to help users, add an early check for the Az module and Az context and emit an instruction (e.g., check `Get-Module -ListAvailable -Name Az` and `Get-AzContext`), so the script fails fast with a helpful message if prerequisites are missing.